golem-worker-executor-base/src/durable_host/filesystem/types.rs

```diff
@@ -484,6 +584,266 @@ impl<Ctx: WorkerCtx> Host for DurableWorkerCtx<Ctx> {
     }
 }
 
+#[async_trait]
+impl<Ctx: WorkerCtx> HostDescriptor for &mut DurableWorkerCtx<Ctx> {
+    fn read_via_stream(
+        &mut self,
+        self_: Resource<Descriptor>,
+        offset: Filesize,
+    ) -> Result<Resource<InputStream>, FsError> {
+        (*self).read_via_stream(self_, offset)
+    }
+
+    fn write_via_stream(
+        &mut self,
+        self_: Resource<Descriptor>,
+        offset: Filesize,
+    ) -> Result<Resource<OutputStream>, FsError> {
+        (*self).write_via_stream(self_, offset)
+    }
+
+    fn append_via_stream(
+        &mut self,
+        self_: Resource<Descriptor>,
+    ) -> Result<Resource<OutputStream>, FsError> {
+        (*self).append_via_stream(self_)
+    }
+
+    async fn advise(
+        &mut self,
+        self_: Resource<Descriptor>,
+        offset: Filesize,
+        length: Filesize,
+        advice: Advice,
+    ) -> Result<(), FsError> {
+        (*self).advise(self_, offset, length, advice).await
+    }
+
+    async fn sync_data(&mut self, self_: Resource<Descriptor>) -> Result<(), FsError> {
+        (*self).sync_data(self_).await
+    }
+
+    async fn get_flags(&mut self, self_: Resource<Descriptor>) -> Result<DescriptorFlags, FsError> {
+        (*self).get_flags(self_).await
+    }
+
+    async fn get_type(&mut self, self_: Resource<Descriptor>) -> Result<DescriptorType, FsError> {
+        (*self).get_type(self_).await
+    }
+
+    async fn set_size(
+        &mut self,
+        self_: Resource<Descriptor>,
+        size: Filesize,
+    ) -> Result<(), FsError> {
+        (*self).set_size(self_, size).await
```

Comment:
Lots of pure delegations here that are adding a lot of boilerplate. What's the purpose of this indirection (again, will look for other implementations to understand better)?